src/olympia/bandwagon/admin.py

```diff
@@ -22,5 +25,34 @@ class CollectionAdmin(admin.ModelAdmin):
     readonly_fields = ('uuid',)
     inlines = (CollectionAddonInline,)
 
+    # Permission checks:
+    # A big part of the curation job for the homepage etc is done through
+    # collections, so users with Admin:Curation can also edit mozilla's
+    # collections as well as their own (allowing them to transfer them to
+    # the mozilla user) through the admin, as well as create new ones.
+    def has_module_permission(self, request):
+        return (
+            acl.action_allowed(request, amo.permissions.ADMIN_CURATION) or
+            super(CollectionAdmin, self).has_module_permission(request))
+
+    def has_change_permission(self, request, obj=None):
+        user = request.user
+        should_allow_curators = (
+            # Changelist, allowed for convenience, should be harmless.
+            obj is None or
+            # Mozilla collection or their own.
+            obj.author and obj.author.username in ('mozilla', user.username))
```

Comment:
Text matching `UserProfile.username`? 🙈  Can we match by `pk` instead please 😉

Reply:
Err yeah :) It's actually `TASK_USER_ID` so it should all work out